test/data.jl

```diff
@@ -271,13 +272,6 @@ module TestData
         v1 = randn(10)
     )
 
-    dt2 = DataTable(
-        a = [:x,:y][[1,2,1,1,2]],
-        b = [:A,:B,:C][[1,1,1,2,3]],
-        v2 = randn(5)
-    )
-    dt2[1,:a] = Nullable()
```

Comment:
This PR currently does not try and handle the case of a user assigning a Nullable() to a typed, non-Nullable() column. We could promote to NullableArray to handle this, but I'm not sure if autopromotion is better than an explicit conversion by the user.